React/Base/RCTEventDispatcher.m

```diff
@@ -45,7 +45,6 @@ @implementation RCTEventDispatcher
   NSMutableArray<NSNumber *> *_eventQueue;
   BOOL _eventsDispatchScheduled;
   NSHashTable<id<RCTEventDispatcherObserver>> *_observers;
-  NSLock *_observersLock;
```

Comment:
This caused a deadlock when updating a view triggers an event. I think simply removing the lock is fine is this case as adding observers is only done when RN is initializing.

Reply:
I am mostly concerned about this part.
Have you tried using recursive lock to avoid the dead lock? That should be much better than removing the lock.